src/main/resources/api/data-repository-openapi.yaml

```diff
@@ -687,6 +687,40 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/ErrorModel'
+  /api/repository/v1/snapshots/{id}/data:
+    get:
+      tags:
+        - snapshots
+        - repository
+      description: Retrieve preview data for a table in a snapshot
+      operationId: lookupSnapshotPreviewById
+      parameters:
+        - $ref: '#/components/parameters/Id'
+        - name: table
+          in: query
+          description: Name of table to get preview data from
+          required: true
+          schema:
+            type: string
+        - name: offset
+          in: query
+          description: The number of rows to skip when retrieving the next preview page
+          schema:
+            type: integer
+            default: 0
```

Comment:
It looks like openapi supports `minimum` and `maximum` for `integer` parameters. If that works, we wouldn't need to validate the minimum values for `offset` and `limit` in the code. See https://swagger.io/docs/specification/describing-parameters/

Reply:
That's a good idea. I'm not sure about offset though. I don't think this should have a lower / upper bound.